### PR TITLE
Serverless build rule engine

### PR DESCRIPTION
**Pull request by olegfomindev9**
### Requirements

Verify and alter the deployment descriptor provided by the developer so to simplify the creation and maintenance of the multi-environment and multi-account serverless realm  

### Description of the Change
The rule based engine has been created
The rule file mimics the original serverless.yml file structure supplying every leaf node with meta information needed to verify and/or populate the field

### Benefits

Forming and validation of the developer supplied serverless.yml file become simple and flexible

### Possible Drawbacks

Complexity of the source code that may hide couple of defects especially in absence of unit tests

### Applicable Issues

So far substitution of functions/name does not work inside the Jenkins environment but it works perfectly well in the groovyConsole that I guess happens because of the difference in groovy versions installed on my box and inside the remote Jenkins

### CODE REVIEW

File: builds/jenkins-build-sls-app/Jenkinsfile [hunk 1]
```diff
@@ -137,14 +138,10 @@ node() {
...
-    try {
-      def dd = prepareServerlessYml(config, environment_logical_id) // Generating the deployment descriptor
-      echo "prepareServerlessYml = ${dd}"
-      writeYaml(file: './serverless.yml', data: dd)
-    } catch(FileAlreadyExistsException faee) {
-      echo 'The serverless.yml exists inside the source code thus disregarding the deployment descriptor content from the service catalog'
-      echoServerlessFile()
-    }
+  /* Writing the file from config */
+    def dd = prepareServerlessYml(config, environment_logical_id) // Generating the deployment descriptor
```
Comment (SiniWilson) [inline builds/jenkins-build-sls-app/Jenkinsfile]: give meaningful variable name

File: builds/jenkins-build-sls-app/Jenkinsfile [hunk 2]
```diff
@@ -6,10 +6,10 @@ import groovy.transform.Field
...
 // To be replaced as @Field def repo_credential_id = "value" for repo_credential_id, repo_base and repo_core
 
-@Field def repo_credential_id
-@Field def repo_base
-@Field def repo_core
-@Field def scm_type
+@Field def repo_credential_id = "jazz_repocreds"
+@Field def repo_base = "slsapp19-gitlab-alb-1957372724.us-east-1.elb.amazonaws.com"
+@Field def repo_core = "slf"
+@Field def scm_type = "gitlab"
```
Comment (SiniWilson) [inline builds/jenkins-build-sls-app/Jenkinsfile]: Remove all hard coded values

File: builds/jenkins-build-sls-app/Jenkinsfile (hunk 1 above)
Comment (olegfomindev9) [reply builds/jenkins-build-sls-app/Jenkinsfile]: I was hoping that acronym 'dd' that stands for 'Deployment Descriptor' is meaningful enough, but obviously I was mistaken :-). I'll rename shortly

File: builds/jenkins-build-sls-app/Jenkinsfile (hunk 2 above)
Comment (olegfomindev9) [reply builds/jenkins-build-sls-app/Jenkinsfile]: That creeps inside over and over again ;-( Removing once again now

File: builds/jenkins-build-sls-app/Jenkinsfile (hunk 1 above)
Comment (olegfomindev9) [reply builds/jenkins-build-sls-app/Jenkinsfile]: Done deal :-)

File: builds/jenkins-build-sls-app/Jenkinsfile (hunk 2 above)
Comment (olegfomindev9) [reply builds/jenkins-build-sls-app/Jenkinsfile]: Done deal :-)

**Pull request merged**